Subject: Card bus's PCI last bus 

Some BIOS assign CardBus bridge to unique last bus number.
We use not the counted number of buses, but the value of PCI BIOS be set as last bus.

Signed-off-by: Go Taniguchi <go@turbolinux.co.jp>
---
---
 drivers/pci/probe.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -775,6 +775,9 @@ int __devinit pci_scan_bridge(struct pci
 		if (child->busn_res.end > max)
 			max = child->busn_res.end;
 	} else {
+#ifdef CONFIG_X86
+		extern int pcibios_last_bus;
+#endif
 		/*
 		 * We need to assign a number to this bus which we always
 		 * do in the second pass.
@@ -795,6 +798,10 @@ int __devinit pci_scan_bridge(struct pci
 		/* Clear errors */
 		pci_write_config_word(dev, PCI_STATUS, 0xffff);
 
+#ifdef CONFIG_X86
+		if(is_cardbus && pcibios_last_bus > 0 && max < pcibios_last_bus)
+			max = pcibios_last_bus;
+#endif
 		/* Prevent assigning a bus number that already exists.
 		 * This can happen when a bridge is hot-plugged, so in
 		 * this case we only re-scan this bus. */
